### ISSUE TITLE
[Bug] Incomplete UIDialog life cycle invocation

### ISSUE THREAD

**Issue by avrahamcool** [adarshpastakia/aurelia-ui-framework#80]
Incomplete UIDialog life cycle invocation - on child elements within the UIDialog.
step to reproduce:
1. create a component that extends UIDialog - call it parent
2. create a component (that implements full life cycle) - call it child
3. require the child component from the parent component - and call it inside the view.
4. activate the modal.

only `created`, `bind`, `attached` life cycle hooks on the child component are called.

`attached` is called explicitly here - but other life cycle hooks are not.
https://github.com/adarshpastakia/aurelia-ui-framework/blob/354ae263ed21e5fd39709b1fdb3d2148070419a8/src/resources/utils/ui-dialog.ts#L78

i'll do my best to create a fix for that and post it soon.

**Comment by avrahamcool** [comment]
I think the best approach is to save a reference to the `childSlot` on the dialog instance.
and call it's full life cycle inside the `closeDialog` function, just like you did with the dialog itself:

**Comment by avrahamcool** [comment]
I'm reopening this issue because I'm not 100% sure about the fix I posted.
I don't fully understand how it works.

regarding the child life cycle:
`bind()` is called automatically, but `attached()` is not (so we call it manually).
and we also need to call `detached()` and `unbind()` manually for some reason.
the strange thing is that although I called `unbind()` the function don't run, and no error is shown.
